### PR TITLE
Add storage features to release notes.

### PR DESCRIPTION
**Pull request by childsb**
Skeleton of storage features for release notes (will update as features solidify)

### CODE REVIEW

File: release-1.8/release_notes_draft.md [hunk 1]
```diff
@@ -44,3 +44,17 @@ please check out the [release notes guidance][] issue.
...
+* Enable containerization of mount dependencies
+* Support Attach/Detach for RWO volumes such as iSCSI, Fibre Channel and RBD
+* Volume Plugin Metrics
+* Snapshots
+* Resizing Volume Support
+* Exposing StorageClass Params To End Users (aka Provisioning configuration in PVC)
+* Mount Options to GA
+* Allow configuration of reclaim policy in StorageClass
+* Expose Storage Usage Metrics
+
```
Comment (rootfs) [inline release-1.8/release_notes_draft.md]: PV spec refactoring for plugins that reference namespaced resources: Azure File, CephFS, iSCSI, Glusterfs

**Comment by jdumars** [comment]
@childsb do you want to add @rootfs point?

**Comment by childsb** [comment]
@jdumars done. thanks @rootfs

**Comment by jdumars** [comment]
@saad-ali LTGY still?

**Pull request merged**